### PR TITLE
idp/keystone: convert fully to quicktest

### PR DESCRIPTION
**Pull request by rogpeppe**
This won't work until the qthttptest PR has merged.

### CODE REVIEW

File: idp/keystone/common_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+package keystone_test
+
+import (
+	"github.com/CanonicalLtd/candid/idp"
+	keystoneidp "github.com/CanonicalLtd/candid/idp/keystone"
+	"github.com/CanonicalLtd/candid/idp/keystone/internal/keystone"
+	"github.com/CanonicalLtd/candid/idp/keystone/internal/mockkeystone"
+	idptest "github.com/CanonicalLtd/candid/idp/qtidptest"
+	candidtest "github.com/CanonicalLtd/candid/internal/qtcandidtest"
+	qt "github.com/frankban/quicktest"
```
Comment (mhilton) [inline idp/keystone/common_test.go]: This should be sorted above the rest

File: idp/keystone/keystone_test.go [hunk 2]
```diff
@@ -15,41 +15,25 @@ import (
...
-	s.server.TokensFunc = testTokens
-	s.server.TenantsFunc = testTenants
-	s.idp = keystoneidp.NewIdentityProvider(s.params)
-	err := s.idp.Init(s.idptest.Ctx, s.idptest.InitParams(c, "https://idp.test"))
-	c.Assert(err, qt.Equals, nil)
+	s.fixture = newFixture(c, fixtureParams{
+		newIDP:      keystoneidp.NewIdentityProvider,
+		tokensFunc:  testTokens,
+		tenantsFunc: testTenants,
+	})
```
Comment (mhilton) [inline idp/keystone/keystone_test.go]: I'm slightly surprised this doesn't need to be closed at the end of the test.

File: idp/keystone/keystone_test.go [hunk 3]
```diff
@@ -216,18 +200,17 @@ identity-providers:
...
-			if test.expectError != "" {
-				c.Assert(err, qt.ErrorMatches, test.expectError)
-				return
-			}
-			c.Assert(err, qt.Equals, nil)
-			c.Assert(conf.IdentityProviders, qt.HasLen, 1)
-			c.Assert(conf.IdentityProviders[0].Name(), qt.Equals, "openstack")
-		})
+	for i, test := range configTests {
+		c.Logf("%d. %s", i, test.about)
```
Comment (mhilton) [inline idp/keystone/keystone_test.go]: I think it's worth converting these to use Run for sub-tests as you go along. Actually why have you removed it?

File: idp/keystone/token_test.go [hunk 4]
```diff
@@ -9,72 +9,52 @@ import (
...
-func (s *tokenSuite) TearDownTest(c *gc.C) {
-	s.Suite.TearDownTest(c)
+func (s *tokenSuite) TestKeystoneTokenIdentityProviderInteractive(c *qt.C) {
+	c.Assert(s.idp.Interactive(), qt.Equals, false)
 }
 
-func (s *tokenSuite) TestKeystoneTokenIdentityProviderInteractive(c *gc.C) {
-	c.Assert(s.idp.Interactive(), gc.Equals, false)
-}
+var logger = loggo.GetLogger("xxxx")
```
Comment (mhilton) [inline idp/keystone/token_test.go]: this looks suspicious.

File: idp/keystone/common_test.go (hunk 1 above)
Comment (rogpeppe) [reply idp/keystone/common_test.go]: done

File: idp/keystone/keystone_test.go (hunk 2 above)
Comment (rogpeppe) [reply idp/keystone/keystone_test.go]: anything that needs to be closed will be Deferred and closed by qtsuite.Run.

File: idp/keystone/keystone_test.go (hunk 3 above)
Comment (rogpeppe) [reply idp/keystone/keystone_test.go]: ha, i think i must have done it and forgot. done.

File: idp/keystone/token_test.go (hunk 4 above)
Comment (rogpeppe) [reply idp/keystone/token_test.go]: removed

**Pull request merged**